src/dune_rules/lib_flags.ml

```diff
@@ -155,8 +162,16 @@ module L = struct
       remove_stdlib dirs ts
   ;;
 
-  let include_flags ?project ts mode =
-    to_iflags (include_paths ?project ts { lib_mode = mode; melange_emit = false })
+  let include_flags ?project ts_direct ts_hidden mode =
+    let hidden_includes =
+      to_hflags
+        (include_paths ?project ts_hidden { lib_mode = mode; melange_emit = false })
+    in
+    let direct_includes =
+      to_iflags
+        (include_paths ?project ts_direct { lib_mode = mode; melange_emit = false })
+    in
+    Command.Args.S [ direct_includes; hidden_includes ]
```

Comment:
Isn't this re-ordering the include flags? e.g. if we have `(libraries x y)`, we'd have `closure(x) closure(y)`, and now we have `x y closure(x) closure(y)`?

Reply:
If I understand things correctly, I don't think anything is being re-ordered; instead some extra `-H` flags are being added at the "end" of the command-line when `implicit_transitive_deps` is `false`:

- If `implicit_transitive_deps` is `true`, then `direct_includes` is `requires_link` (from `Compilation_context`), and `hidden_includes` is empty; this is the same as today, and nothing changes.
- If `implicit_transitive_deps` is `false` (and OCaml is recent enough, etc) `direct_includes` is `requires_compile` (from `Compilation_context`), and `hidden_includes` contains some extra flags relative to today, but the order of `direct_includes` does not change.

Do you agree with this understanding @MA0010?